mockgcp/mockspanner/database.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockspanner
+
+import (
+	"context"
+	"strings"
+
+	"cloud.google.com/go/longrunning/autogen/longrunningpb"
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/spanner/admin/database/v1"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+var _ pb.DatabaseAdminServer = &SpannerDatabaseV1{}
+
+type SpannerDatabaseV1 struct {
+	*MockService
+	pb.UnimplementedDatabaseAdminServer
+}
+
+func (s *SpannerDatabaseV1) GetDatabase(ctx context.Context, req *pb.GetDatabaseRequest) (*pb.Database, error) {
+	name, err := s.parseDatabaseName(req.GetName())
+	if err != nil {
+		return nil, err
+	}
+	fqn := name.String()
+
+	obj := &pb.Database{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+func (s *SpannerDatabaseV1) CreateDatabase(ctx context.Context, req *pb.CreateDatabaseRequest) (*longrunningpb.Operation, error) {
+	databaseName := databaseNameFromCreateStatement(req.GetCreateStatement())
+	fqn := req.GetParent() + "/databases/" + databaseName
+
+	obj := &pb.Database{}
+	obj.Name = fqn
+
+	now := timestamppb.Now()
+	obj.CreateTime = now
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	return s.operations.NewLRO(ctx)
+}
+
+// Given a createStatement string of the format "CREATE DATABASE `<dbname>`",
+// return the database name as a string.
+func databaseNameFromCreateStatement(createStatement string) string {
+	dbname := strings.TrimPrefix(createStatement, "CREATE DATABASE `")
+	dbname = strings.TrimSuffix(dbname, "`")
+	return dbname
```

Comment:
You may to raise an error if e.g. this name has any spaces in it, just to catch the case where the create statement is unexpected ... but just an idea to save us time in the future (it doesn't have to be perfect, but because it's a mock we can just klog.Fatalf ... I _didn't_ do that for our service enablement mock and it bites me every time ... must fix that!)

Reply:
Nice, thanks. Added a couple checks for valid db name